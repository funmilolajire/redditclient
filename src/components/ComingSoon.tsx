import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';

export const ComingSoon = () => {
    const boxRef = useRef<HTMLDivElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({ repeatDelay: 2 })
        tl.from([boxRef.current], {
            y: '-55vh',
            duration: 3,
            delay: 2,
            opacity: 0,
            ease: 'elastic'
        })
    })

    return (
        <div ref={boxRef} className="container flex items-center justify-center h-screen font-extrabold ComingSoon">
            <div className="bg-yellow-500 border-2 border-gray-800 border-solid skew-y-2 transform-gpu text-gray-50 text-9xl">
                <p>Coming Soon</p>
            </div>
        </div>
    )
}