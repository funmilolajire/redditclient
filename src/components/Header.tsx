import { useRef, useLayoutEffect } from 'react';
import { gsap } from 'gsap';
import logo from '../images/logo192.png';

export const Header = () => {
    const boxRef = useRef<HTMLDivElement>(null);
    const logoRef = useRef<HTMLImageElement>(null);
    const titleRef = useRef<HTMLHeadingElement>(null);

    useLayoutEffect(() => {
        const tl = gsap.timeline({});
        tl.from([boxRef.current], {
            duration: 1
        })
        tl.from([logoRef.current], {
            x: '-3000px',
            opacity: 0,
            ease: 'circ'
        })
        tl.from([titleRef.current], {
            x: '3000px',
            opacity: 0,
            ease: 'slow'
        })
    })

    return (
        <header>
            <div className="header-inner">
                <section ref={boxRef}>
                    <figure ref={logoRef}>
                        <img src={logo} alt="site logo" />
                    </figure>
                    <h1 ref={titleRef}>
                        <span>Reddit</span>Client
                    </h1>
                </section>
            </div>
        </header>
    )
}