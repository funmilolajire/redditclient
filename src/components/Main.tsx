import { Listings } from './Listings';
import { Posts } from './Posts';
export const Main = () => (
    <div className="Main">
        <Listings />
        <Posts />
    </div>
)