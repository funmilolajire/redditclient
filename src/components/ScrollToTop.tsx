import { useLayoutEffect, useCallback, useState } from 'react';
import { IoChevronUpCircle } from 'react-icons/io5';

export const ScrollToTop = () => {
    const [showScroll, setShowScroll] = useState(false)

    const checkScrollTop = useCallback(() => {
        if (!showScroll && window.pageYOffset > 400) {
            setShowScroll(true)
        } else if (showScroll && window.pageYOffset <= 400) {
            setShowScroll(false)
        }
    }, [showScroll, setShowScroll]);

    const buttonStyle = {
        display: showScroll ? 'flex' : 'none',
    }

    const scrollTop = () => {
        window.scrollTo({
            top: 0,
            behavior: 'smooth'
        })
    }

    useLayoutEffect(() => {
        window.addEventListener('scroll', checkScrollTop)
        return () => {
            window.removeEventListener('scroll', checkScrollTop)
        }
    }, [checkScrollTop])

    return (
        <span className="ScrollToTop" title="scroll to top" aria-label="scroll to top">
            <IoChevronUpCircle className="ScrollToTopButton" role="button" data-testid="ScrollToTopButton" style={buttonStyle} onClick={scrollTop} />
        </span>
    )
}