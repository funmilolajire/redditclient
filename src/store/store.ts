import { configureStore } from '@reduxjs/toolkit';
import termReducer from './slices/termSlice';

export default configureStore({
  reducer: {
    term: termReducer,
  },
});
