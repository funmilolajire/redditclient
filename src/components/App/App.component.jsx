import { Header } from '../Header/Header.component';
import { ComingSoon } from '../ComingSoon/ComingSoon.component';

export const App = () => {
    return (
        <div className="container max-h-screen overflow-hidden App">
            <Header />
            <ComingSoon />
        </div>
    )
}