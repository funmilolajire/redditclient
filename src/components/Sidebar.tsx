import { useRef, useLayoutEffect, useEffect } from 'react';
import { Searchbar } from "./Searchbar";
import { Subreddits } from './Subreddits';
import { gsap } from 'gsap';
import { HiOutlineMenuAlt2 } from 'react-icons/hi';
import { IoCloseOutline } from 'react-icons/io5';
import { useAppSelector } from '../store/hooks';

export const Sidebar = () => {
    const boxRef = useRef<HTMLDivElement>(null);
    const menuRef = useRef<HTMLDivElement>(null);
    useLayoutEffect(() => {
        const tl = gsap.timeline({})
        tl.from([boxRef.current], {
            y: '-120%',
            delay: 0,
            duration: 2,
            ease: 'power3.inOut'
        })
        tl.from([menuRef.current], {
            y: '-120%',
            delay: 0,
            duration: 2,
            ease: 'power3.inOut'
        })
    }, [])

    const openMobileMenu = () => {
        const mobileMenu = menuRef.current?.dataset.mobileMenu
        if (mobileMenu) {
            menuRef.current?.classList.remove('sm:flex')
            boxRef.current?.classList.remove('sm:hidden')
        }
    }

    const closeMobileMenu = () => {
        const menu = boxRef.current?.dataset.menu
        if (menu) {
            boxRef.current?.classList.add('sm:hidden')
            menuRef.current?.classList.add('sm:flex')
        }
    }

    const subreddit = useAppSelector(state => state.posts.subreddit)
    const listing = useAppSelector(state => state.posts.listing);
    const term = useAppSelector(state => state.posts.term);
    const showModal = useAppSelector(state => state.moreInformation.showModal);

    useEffect(() => {
        closeMobileMenu()
    }, [subreddit, listing, term, showModal])

    return (
        <>
            <aside ref={boxRef} data-menu className="Sidebar sm:hidden">
                <div className="Sidebar--Container">
                    <span onClick={closeMobileMenu} className="close"><IoCloseOutline /> </span>
                    <Searchbar />
                    <Subreddits />
                </div>
            </aside>

            <aside ref={menuRef} data-mobile-menu onClick={openMobileMenu} className="Mobile--Sidebar sm:flex hidden">
                <span className="menu"><HiOutlineMenuAlt2 /> </span>
            </aside>
        </>
    )
}