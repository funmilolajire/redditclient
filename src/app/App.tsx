import { Header } from '../components/Header';
import { Main } from '../components/Main';
import { Sidebar } from '../components/Sidebar';
import { ScrollToTop } from '../components/ScrollToTop';
import '../scss/main.scss';

export const App = () => (
    <div className="App">
        <Header />
        <main>
            <Sidebar />
            <Main />
        </main>
        <ScrollToTop />
    </div>
)