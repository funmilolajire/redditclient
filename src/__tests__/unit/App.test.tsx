import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import { App } from '../../app/App';
import { Header } from '../../components/Header';
import { Main } from '../../components/Main';
import { Sidebar } from '../../components/Sidebar';
import { ScrollToTop } from '../../components/ScrollToTop';
// import { ComingSoon } from '../../components/ComingSoon';

window.scrollTo = jest.fn()

describe('renders App Component', () => {
  const AppComponent = mount(
    <Provider store={store}>
      <App />
    </Provider>
  );
  afterEach(() => {
    jest.resetAllMocks();
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
  describe('make snapshot of App component', () => {
    test('matches the snapshot', () => {
      expect(AppComponent).toMatchSnapshot();
    })
  })


  describe('Contains Children Components', () => {
    test('contains all components', () => {
      // const getComponents = AppComponent.containsAllMatchingElements([<Header />, <ComingSoon />])
      const getComponents = AppComponent.containsAllMatchingElements([<Header />, <Main />, <Sidebar />, <ScrollToTop />]);
      expect(getComponents).toBeTrue();
    })
  })

});
