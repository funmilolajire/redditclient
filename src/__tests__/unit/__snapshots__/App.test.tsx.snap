// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component make snapshot of App component matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div
      className="App"
    >
      <Header>
        <header>
          <div
            className="header-inner"
          >
            <section>
              <figure>
                <img
                  alt="site logo"
                  src="logo192.png"
                />
              </figure>
              <h1>
                <span>
                  Reddit
                </span>
                Client
              </h1>
            </section>
          </div>
        </header>
      </Header>
      <main>
        <Sidebar>
          <aside
            className="Sidebar sm:hidden"
            data-menu={true}
          >
            <div
              className="Sidebar--Container"
            >
              <span
                className="close"
                onClick={[Function]}
              >
                <IoCloseOutline>
                  <IconBase
                    attr={
                      Object {
                        "viewBox": "0 0 512 512",
                      }
                    }
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M368 368L144 144m224 0L144 368"
                        fill="none"
                        key="0"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="32"
                      />
                    </svg>
                  </IconBase>
                </IoCloseOutline>
                 
              </span>
              <Searchbar>
                <div
                  className="Searchbar"
                  onKeyDown={[Function]}
                >
                  <div
                    className="Searchbar--Container"
                  >
                    <input
                      autoComplete="off"
                      id="search"
                      name="search"
                      onChange={[Function]}
                      placeholder="search"
                      type="text"
                      value=""
                    />
                    <div
                      className="Searchbar--Icon"
                      onClick={[Function]}
                    >
                      <svg
                        height="18"
                        viewBox="0 0 18 18"
                        width="18"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clipRule="evenodd"
                          d="M12.5 11H11.7L11.4 10.7C12.4 9.6 13 8.1 13 6.5C13 2.9 10.1 0 6.5 0C2.9 0 0 2.9 0 6.5C0 10.1 2.9 13 6.5 13C8.1 13 9.6 12.4 10.7 11.4L11 11.7V12.5L16 17.5L17.5 16L12.5 11ZM6.5 11C4 11 2 9 2 6.5C2 4 4 2 6.5 2C9 2 11 4 11 6.5C11 9 9 11 6.5 11Z"
                          fillRule="evenodd"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </Searchbar>
              <Subreddits>
                <section
                  className="pt-12 Subreddits"
                >
                  <h2
                    className=""
                  >
                    subreddits
                  </h2>
                  <ul
                    className="subreddits--list"
                  />
                </section>
              </Subreddits>
            </div>
          </aside>
          <aside
            className="Mobile--Sidebar sm:flex hidden"
            data-mobile-menu={true}
            onClick={[Function]}
          >
            <span
              className="menu"
            >
              <HiOutlineMenuAlt2>
                <IconBase
                  attr={
                    Object {
                      "fill": "none",
                      "stroke": "currentColor",
                      "viewBox": "0 0 24 24",
                    }
                  }
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4 6h16M4 12h16M4 18h7"
                      key="0"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                    />
                  </svg>
                </IconBase>
              </HiOutlineMenuAlt2>
               
            </span>
          </aside>
        </Sidebar>
        <Main>
          <div
            className="Main"
          >
            <Listings>
              <div
                className="Listings"
              >
                <ul
                  className="Listings--List"
                >
                  <Listing
                    key="hot"
                    listItem="hot"
                  >
                    <li
                      data-testid="hot"
                      id="hot"
                      onClick={[Function]}
                    >
                      hot
                    </li>
                  </Listing>
                  <Listing
                    key="new"
                    listItem="new"
                  >
                    <li
                      data-testid="new"
                      id="new"
                      onClick={[Function]}
                    >
                      new
                    </li>
                  </Listing>
                  <Listing
                    key="top"
                    listItem="top"
                  >
                    <li
                      data-testid="top"
                      id="top"
                      onClick={[Function]}
                    >
                      top
                    </li>
                  </Listing>
                  <Listing
                    key="best"
                    listItem="best"
                  >
                    <li
                      data-testid="best"
                      id="best"
                      onClick={[Function]}
                    >
                      best
                    </li>
                  </Listing>
                  <Listing
                    key="rising"
                    listItem="rising"
                  >
                    <li
                      data-testid="rising"
                      id="rising"
                      onClick={[Function]}
                    >
                      rising
                    </li>
                  </Listing>
                  <Listing
                    key="controversial"
                    listItem="controversial"
                  >
                    <li
                      data-testid="controversial"
                      id="controversial"
                      onClick={[Function]}
                    >
                      controversial
                    </li>
                  </Listing>
                </ul>
              </div>
            </Listings>
            <Posts>
              <div
                className="Posts"
              />
            </Posts>
          </div>
        </Main>
      </main>
      <ScrollToTop>
        <span
          aria-label="scroll to top"
          className="ScrollToTop"
          title="scroll to top"
        >
          <IoChevronUpCircle
            className="ScrollToTopButton"
            data-testid="ScrollToTopButton"
            onClick={[Function]}
            role="button"
            style={
              Object {
                "display": "none",
              }
            }
          >
            <IconBase
              attr={
                Object {
                  "viewBox": "0 0 512 512",
                }
              }
              className="ScrollToTopButton"
              data-testid="ScrollToTopButton"
              onClick={[Function]}
              role="button"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <svg
                className="ScrollToTopButton"
                data-testid="ScrollToTopButton"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                role="button"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                    "display": "none",
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48zm107.31 259.31a16 16 0 01-22.62 0L256 222.63l-84.69 84.68a16 16 0 01-22.62-22.62l96-96a16 16 0 0122.62 0l96 96a16 16 0 010 22.62z"
                  key="0"
                />
              </svg>
            </IconBase>
          </IoChevronUpCircle>
        </span>
      </ScrollToTop>
    </div>
  </App>
</Provider>
`;
